Update documentation, comments, metadata to reference GitHub as canonical repository

Grinnz opened the pull request:
These changes are primarily to POD and comments.

The "Merging from a branch via GitHub" section is removed because it's not really relevant when GitHub is the standard repository.

I'm not sure what to do with the "A note on camel and dromedary" section in perlgit or the "Read access via rsync" section in perlhack.

References to cloning via HTTP being "slower" were removed as I see no evidence of this with the GitHub repository.

Additional updates are needed to reference the GitHub pull request workflow, I may work on this in a future patch.

Updating documentation to reference using GitHub to submit bug reports (other than the metadata entry) is not part of this patch and will be a significant undertaking that needs to start with the perlbug utility.

Grinnz committed:
Update repository and bugtracker URLs to GitHub in makemeta

regen META files

Grinnz committed:
Update POD and comments to reference GitHub as canonical repository

Grinnz commented:
Forgot to mention, this should obviously wait until the migration is complete.

atoomic commented:
I think we should remove the `A note on camel and dromedary` section from `pod/perlgit.pod`

Corion commented:
This will also need the additional fix to `Porting/corelist.pl`, where a check for upstream / bug tracker rt.perl.org is made. I guess you'll want to add whatever becomes the canonical upstream URL there as well, like maybe `https://github.com/Perl/perl5/issues`:

```
  next if $bug_tracker eq "'https://github.com/Perl/perl5/issues'";
```

Grinnz committed:
Update Porting/corelist.pl to recognize GitHub issue tracker

Grinnz commented:
@Corion added, thanks

Grinnz committed:
perlgit: remove "A note on camel and dromedary"

Grinnz commented:
Removed the perlgit section on camel and dromedary

toddr commented on pod/perlgit.pod:
```diff
@@ -814,35 +814,6 @@ original commit in the new commit message.
 Before pushing any change to a maint version, make sure you've
 satisfied the steps in L</Committing to blead> above.
 
-=head2 Merging from a branch via GitHub
```
This is cool but we probably need a TODO to add something about PRs in github

Grinnz replied:
I would like to do this as a separate PR, the references to perlbug are extensive.

toddr commented on pod/perlgit.pod:
```diff
@@ -853,7 +824,7 @@ Fortunately, there is a way to get your change smoke-tested on various
 OSes: push it to a "smoke-me" branch and wait for certain automated
 smoke-testers to report the results from their OSes.
 A "smoke-me" branch is identified by the branch name: specifically, as
```
Eventually pull requests will smoke.

Grinnz committed:
perlgit: Remove redundant 'Committing your changes' section

Grinnz commented:
Removed the 'Committing your changes' section from perlgit since it contains the same information as the preceding section, aside from a link to perlhack that I added to the preceding section.
